src/main/java/org/aerogear/kryptowire/workflow/KWSubmitStep.java

```diff
@@ -95,12 +105,22 @@ protected Void run() throws Exception {
             listener.getLogger().println("kwSubmit: " + step.platform + " : " + step.filePath);
             KryptowireService kws = new KryptowireServiceImpl(kwEndpoint,  kwApiKey);
             JSONObject resp = kws.submit(step.platform, fp);
+
+            String uuid = resp.getString("uuid");
+            String platform = resp.getString("platform");
+            String pkg = resp.getString("package");
+            String version = resp.getString("version");
+            String hash = resp.getString("hash");
+
             listener.getLogger().println("kw msg: " + resp.get("msg"));
-            listener.getLogger().println("kw uuid: " + resp.get("uuid"));
-            listener.getLogger().println("kw platform: " + resp.get("platform"));
-            listener.getLogger().println("kw package: " + resp.get("package"));
-            listener.getLogger().println("kw version: " + resp.get("version"));
-            listener.getLogger().println("kw hash: " + resp.get("hash"));
+            listener.getLogger().println("kw uuid: " + uuid);
+            listener.getLogger().println("kw platform: " + platform);
+            listener.getLogger().println("kw package: " + pkg);
+            listener.getLogger().println("kw version: " + version);
+            listener.getLogger().println("kw hash: " + hash);
+
+            BinaryInfo info = new BinaryInfo(pkg, uuid, platform, version, hash);
+            build.addAction(new BinaryHistoryAction(info));
```

Comment:
Might be good, at some point, to just have a method on the service or a wrapper on it that returns the Java object (BinaryInfo) for the api responses rather than converting them here.